specs/Print.md

```diff
@@ -0,0 +1,473 @@
+    CHECK_FEATURE_RETURN(webView2_15);
+
+    wil::com_ptr<ICoreWebView2Environment6> webviewEnvironment6;
+    CHECK_FAILURE(m_webViewEnvironment->QueryInterface(IID_PPV_ARGS(&webviewEnvironment6)));
+    CHECK_FEATURE_RETURN(webviewEnvironment6);
+
+    wil::com_ptr<ICoreWebView2PrintSettings> printSettings = nullptr;
+    CHECK_FAILURE(webviewEnvironment6->CreatePrintSettings(&printSettings));
+
+    wil::com_ptr<ICoreWebView2PrintSettings2> printSettings2;
+    CHECK_FAILURE(printSettings->QueryInterface(IID_PPV_ARGS(&printSettings2)));
+    CHECK_FEATURE_RETURN(printSettings2);
+
+    CHECK_FAILURE(printSettings2->put_Orientation(COREWEBVIEW2_PRINT_ORIENTATION_PORTRAIT));
+    CHECK_FAILURE(printSettings2->put_ShouldPrintBackgrounds(true));
+
+    CHECK_FAILURE(
+        webView2_15->PrintToPdfStream(
+            printSettings2.get(),
+            Callback<ICoreWebView2PrintToPdfStreamCompletedHandler>(
+                [this](HRESULT errorCode, IStream* pdfData) -> HRESULT
+                {
+                    CHECK_FAILURE(errorCode);
+                    DisplayPdfDataInPrintDialog(pdfData);
+                    return S_OK;
+                })
+                .Get()));
+     return true;
+}
+```
+
+```c#
+async void PrintToPdfStream()
+{
+    CoreWebView2PrintSettings printSettings = null;
+    printSettings = WebViewEnvironment.CreatePrintSettings();
+    printSettings.Orientation = CoreWebView2PrintOrientation.Portrait;
+    printSettings.ShouldPrintBackgrounds = true;
+
+    MemoryStream pdfStream = new MemoryStream();
+    System.IO.Stream stream = await webView.CoreWebView2.PrintToPdfStreamAsync(printSettings);
+    DisplayPdfDataInPrintDialog(pdfData);
+}
+
+// Function to display current page pdf data in a custom print preview dialog.
+void DisplayPdfDataInPrintDialog(Stream pdfData)
+{
+    // You can display the printable pdf data in a custom print preview dialog to the end user.
+}
+```
+
+# API Details
+
+## Win32 C++
+``` cpp
+/// Specifies the print dialog kind.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_DIALOG_KIND {
+  /// Opens the browser print preview dialog.
+  COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER_PRINT_PREVIEW,
+
+  /// Opens the system print dialog.
+  COREWEBVIEW2_PRINT_DIALOG_KIND_SYSTEM_PRINT,
+
+} COREWEBVIEW2_PRINT_DIALOG_KIND;
+
+/// Specifies the printing side.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_SIDE {
+  /// Single-sided printing.
+  COREWEBVIEW2_PRINT_SIDE_SINGLE,
+
+  /// Double-sided, horizontal printing.
+  COREWEBVIEW2_PRINT_SIDE_BOTH_HORIZONTAL,
+
+  /// Double-sided, vertical printing.
+  COREWEBVIEW2_PRINT_SIDE_BOTH_VERTICAL,
+
+} COREWEBVIEW2_PRINT_SIDE;
+
+[uuid(92269C88-1CC1-47E0-9A87-71458409BA15), object, pointer_default(unique)]
+interface ICoreWebView2_15 : ICoreWebView2_14 {
+  /// Print the current page asynchronously to the specified printer with the provided settings.
+  /// See `ICoreWebView2PrintSettings2` for description of settings. The handler will return `E_ABORT`
```

Comment:
Returning an error implies that it's an app bug. Do we need to worry about timing conditions where a printer disappears? (Thus this should be a Try?)

Reply:
ERROR_PRINTER_DELETED
ERROR_PRINTER_NOT_FOUND
ERROR_INVALID_PRINTER_NAME